utils/validate_pr_quickstarts.js

```diff
@@ -0,0 +1,359 @@
+  $id: ID!
+  $quickstartMetadata: Nr1CatalogQuickstartMetadataInput!
+) {
+    nr1CatalogSubmitQuickstart(
+      dryRun: true
+      id: $id
+      quickstartMetadata: $quickstartMetadata
+    ) {
+        quickstart {
+          id
+        }
+      }
+  }
+`;
+
+/**
+ * Gets the quickstart portion of a given file path.
+ * @param {String} filePath - Full file path of a file in a quickstart.
+ * @param {String} targetChild - Node in file path that should be preceded by a quickstart directory.
+ * @return {String} Node in file path of the quickstart.
+ */
+const getQuickstartNode = (filePath, targetChild) => {
+  const splitFilePath = filePath.split('/');
+  return splitFilePath[
+    splitFilePath.findIndex((path) => path === targetChild) - 1
+  ];
+};
+
+/**
+ * Identifies where in a given file path to look for a quickstart directory.
+ * @param {String} filePath - Full file path of a file in a quickstart.
+ * @return {Function|undefined} Called function with arguments to determine the quickstart of a given file path.
+ */
+const getQuickstartFromFilename = (filePath) => {
+  if (!filePath.includes('quickstarts/')) {
+    return undefined;
+  }
+
+  if (filePath.includes('/alerts/')) {
+    return getQuickstartNode(filePath, 'alerts');
+  }
+
+  if (filePath.includes('/dashboards/')) {
+    return getQuickstartNode(filePath, 'dashboards');
+  }
+
+  if (filePath.includes('/images/')) {
+    return getQuickstartNode(filePath, 'images');
+  }
+
+  const targetChildNode = filePath.split('/').pop();
+
+  return getQuickstartNode(filePath, targetChildNode);
+};
+
+/**
+ * Looks up corresponding quickstart config files for quickstarts known to have changes in a PR.
+ * @param {Set} quickstartDirectories - Set of unique quickstart directories.
+ * @return {Array} Collection of config file paths.
+ */
+const getQuickstartConfigPaths = (quickstartDirectories) => {
+  const allQuickstartConfigPaths = findMainQuickstartConfigFiles();
+
+  return [...quickstartDirectories].reduce((acc, quickstartDirectory) => {
+    const match = allQuickstartConfigPaths.find((path) =>
+      path.split('/').includes(quickstartDirectory)
+    );
+    if (match) {
+      acc.push(match);
+    }
+    return acc;
+  }, []);
+};
+
+/**
+ * Builds input argument for submitQuickstart GraphQL mutation.
+ * @param {Object} quickstartConfig - An object containing the path and contents of a quickstart config file.
+ * @return {Object} An object that represents a quickstart in the context of a GraphQL mutation.
+ */
+const buildMutationVariables = (quickstartConfig) => {
+  const {
+    authors,
+    categoryTerms,
+    description,
+    title,
+    documentation,
+    logo,
+    keywords,
+    summary,
+    installPlans,
+    id,
+  } = quickstartConfig.contents[0] || {};
+  const alertConfigPaths = getQuickstartAlertsConfigs(quickstartConfig.path);
+  const dashboardConfigPaths = getQuickstartDashboardConfigs(
+    quickstartConfig.path
+  );
+
+  return {
+    id: id,
```

Comment:
Pretty sure a brand new quickstart will not have an  `id` in its `config.yml`.  Seeing the addition of a UUID on the `release` workflow, so at the workflow  of candidate PR validation we wouldn't expect an ID: https://github.com/newrelic/newrelic-quickstarts/blob/main/.github/workflows/release.yml#L14. 

That means this `id` value will be `undefined` when sending a validation mutation, so there will always be an error that will make the action / workflow fail on a new quickstart...

It will be good to smoke test this and confirm once the `submitQuickstart` mutation is deployed. 

If the above is correct, we can find a way to handle. Holding off on adding anything preemptively for now.

Reply:
did some quick testing in staging. we can't supply empty, or an id with an invalid "shape" -- looks like it has to be guid / uuid format. however, as long as we do that, the id can be any value and we can validate with it.

so the process would be:
* grab id from quickstart
* if its new and id doesn't exist, replace it with a default id. could be `00000000-0000-0000-0000-000000000000` as an example
* validate using that as the id in the query

Reply:
Got it, I've implemented a fallback UUID here for validation of new quickstarts: c2703ffc

Thinking out loud through an edge case / tradeoff: With this change, any modification to an existing quickstart config where someone has accidentally removed the assigned ID in the `config.yml` would not benefit from validation of that missing field. Suppose in that case a brand new quickstart would be made if the PR is merged, which seems reasonable. I don't see how this PR workflow could know if a given quickstart without an `id` in its `config.yml` is a new addition as opposed to a modified existing config.

Feels okay to move forward with the mock UUID for validation purposes.

Reply:
talked with @zstix and it sounds like we're cool with this for now. i created [a ticket](https://github.com/newrelic/newrelic-quickstarts/issues/698) for this edge case so we can keep track of it.

Reply:
Great, thank you!